packcheck.sh

```diff
@@ -1277,8 +1328,46 @@ create_and_unpack_pkg_dist() {
   cd .packcheck || exit 1
   test "${tarpath:0:1}" == / || tarpath=../$tarpath
   $OS_UNGZTAR_CMD $tarpath
+  echo
 
   cd ${1}
+  if test "$BUILD" = stack
+  then
+    test -e stack.yaml \
+      && echo "WARNING! Avoid including stack.yaml in a distribution" \
+      && run_verbose rm -f stack.yaml
+    $STACKCMD init
```

Comment:
Why?

PS: I haven't read the whole code. Is it required to understand the context?

Reply:
When a build tool builds a dependency it won't take an existing stack.yaml or cabal.project into account. So our testing here would be wrong, because such a project file/stack config may have a source-repository-package for example.